cmd/kube-scheduler/app/server.go

```diff
@@ -58,8 +58,11 @@ import (
 	"k8s.io/klog"
 )
 
-// NewSchedulerCommand creates a *cobra.Command object with default parameters
-func NewSchedulerCommand() *cobra.Command {
+// PluginOption is an function type used for dealing with scheduler registry.
+type PluginOption func(framework.Registry) error
```

Comment:
PluginOption is a bit confusing. I don't have a great suggestion for the name. Does `PluginRegisterer` make more sense?

Reply:
The only exported things in this package are `NewSchedulerCommand` and `Run`. I suggest that the option type be called `Option`.

Reply:
I am not good at naming. From my perspective, `Option` is most general that it could deal with every things to `framework.Registry`, `PluginOption` represents that it deals with registry plugin related things, `PluginRegisterer` represents that it registries plugin to `framework.Registry`, that is what `WithPlugin` is doing now.

Scheduler also has [`Option`](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/scheduler.go#L75). If we'll add some thing more `WithXXX` for `framework.Registry`, it might be better to use `Option`, otherwise either one is OK for me.

Reply:
> Scheduler also has Option. If we'll add some thing more WithXXX for framework.Registry, it might be better to use Option

Because of this reason, I renamed `PluginOption` to `Option` in the latest commit.  @baasbank 
 Does it make sense for you?  PTAL